### PR TITLE
Fix UX Issues

### PR DESCRIPTION
**Pull request by Gauthamastro**
## Describe your changes
1. Solves https://github.com/Polkadex-Substrate/orderbook/issues/872
2. Swap deposit tokens for ED in PDEX automatically
3. Ability to pay withdrawal fees in tokens 
4. Claim callable by anyone.

## Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [x] I removed all Clippy and Formatting Warnings. 
- [x] I added required Copyrights.

### CODE REVIEW

**Comment by Gauthamastro** [comment]
@zktony to add test cases for 

- [ ] Swap for ED during deposit
- [ ] Swapping for withdrawal fees in Thea
- [ ] Check the situation when swap fails 
- [ ] Check the situation when the full deposit/withdrawal amount is used by the swap

File: pallets/ocex/src/lib.rs [hunk 1]
```diff
@@ -437,8 +385,9 @@ pub mod pallet {
 		pub fn add_proxy_account(origin: OriginFor<T>, proxy: T::AccountId) -> DispatchResult {
 			let main_account = ensure_signed(origin)?;
 			ensure!(Self::orderbook_operational_state(), Error::<T>::ExchangeNotOperational);
-			// TODO: Avoid duplicate Proxy accounts
 			ensure!(<Accounts<T>>::contains_key(&main_account), Error::<T>::MainAccountNotFound);
+			// Avoid duplicate Proxy accounts
+			ensure!(!<Proxies<T>>::contains_key(&proxy), Error::<T>::ProxyAlreadyRegistered);
```
Comment (serhii-temchenko) [inline pallets/ocex/src/lib.rs:390]: After adding this assertion - benchmark ([this](https://github.com/Polkadex-Substrate/Polkadex/blob/28e7a135e932e0fecdd14508fddc1f30b1e45e58/pallets/ocex/src/benchmarking.rs#L92-L109)) fails on calling `add_proxy_account` extrinsic with error:
> Error: Input("Benchmark pallet_ocex_lmp::add_proxy_account failed: ProxyAlreadyRegistered")

Please update benchmark logic as well.

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

File: pallets/ocex/src/lib.rs (hunk 1 above)
Comment (Gauthamastro) [reply pallets/ocex/src/lib.rs:390]: Resolved

**Comment by serhii-temchenko** [comment]
CI is still failing, please take a look.

**Comment by serhii-temchenko** [comment]
> @zktony to add test cases for
> 
> * [ ]  Swap for ED during deposit
> * [ ]  Swapping for withdrawal fees in Thea
> * [ ]  Check the situation when swap fails
> * [ ]  Check the situation when the full deposit/withdrawal amount is used by the swap

I assume this will be resolved in scope of the issue #881 and in different PR.
Approving.

**Review by serhii-temchenko — APPROVED** [review]

**Pull request merged**